content/blog/2025/02/github-copilot-in-espressif-ide/index.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "GitHub Copilot Now in Espressif-IDE with Copilot4Eclipse"
+date: 2025-02-05
+showAuthor: false
+disableComments: false
+tags: ["Espressif-IDE", "GitHub Copilot", "IDE"]
+authors:
+- "kondal-kolipaka"
+---
+
+We all love [GitHub Copilot](https://github.com/features/copilot) as a powerful AI-powered coding assistant that enhances productivity. However, until now, it has been officially available only for VS Code and a few other editors. But here’s some good news for Eclipse users! We have tried the [Copilot4Eclipse](https://www.genuitec.com/products/copilot4eclipse/) plugin with the [Espressif-IDE C/C++ Editor](https://docs.espressif.com/projects/espressif-ide/en/latest/additionalfeatures/lspeditor.html#lsp-c-c-editor), and it works amazingly well, helping to write code faster and fix bugs.
+
+## Introducing Copilot4Eclipse
+
+If you've been waiting for GitHub Copilot support in Eclipse, look no further than [Copilot4Eclipse](https://www.genuitec.com/products/copilot4eclipse/) — a fantastic plugin that brings Copilot's AI-assisted code generation to the Eclipse IDE. While the experience may not be identical to VS Code, Copilot4Eclipse offers similar features, making AI-assisted development possible within Eclipse.
+
+## How Copilot4Eclipse Enhances Development
+
+### AI-Powered Code Suggestions
+
+Copilot4Eclipse generates entire functions, suggests code completions, and refactors snippets, streamlining development and boosting efficiency. It assists with boilerplate code, ESP32 API calls, and even offers intelligent improvements to your implementation—helping you write better code, faster.
+
+{{< figure default=true src="assets/copilot-with-espressif-ide.webp" >}}
+
+### Seamless Integration with Espressif-IDE
+
+Once you install Copilot4Eclipse in Espressif-IDE, you’ll see the Copilot menu, where you can enable GitHub Copilot suggestions. We tested Copilot4Eclipse with Espressif-IDE on ESP-IDF projects, and it worked flawlessly!
+
+{{< figure default=true src="assets/copilot-menu.webp" >}}
+
+### GitHub Copilot Chat
+
+GitHub Copilot Chat is an AI-powered coding assistant that provides real-time guidance directly within your IDE. It allows developers to choose a file, ask questions, get code explanations, debug errors, and generate code snippets—all through an interactive chat interface.
+
+{{< figure default=true src="assets/copilot-chat.webp" >}}
+
+### Supports Various Languages
+
+Whether you're coding in C or C++, GitHub Copilot’s AI-driven suggestions can help you write better code, faster.
+
+{{< figure default=true src="assets/preferences.webp" >}}
+
+## Getting Started with Copilot4Eclipse
+
+- **Install the [Copilot4Eclipse](https://www.genuitec.com/products/copilot4eclipse/docs/installation) Plugin** - Download and install the Copilot4Eclipse extension from the [Eclipse Marketplace](https://marketplace.eclipse.org/content/copilot4eclipse).
```

Comment:
I suggest adding a paragraph before the bullet points.

`To get started with Copilot4Eclipse you will need to:`